clash.cabal install from snap fails due to missing libraries

I attempted to use the clash snap (version 1.6.2) on Ubuntu 20.04 to install a cabal package, but it fails due to missing libraries:

```
$ clash.cabal install strict
Resolving dependencies...
Build profile: -w ghc-8.6.5 -O1
In order, the following will be built (use -v for more details):
 - base-orphans-0.8.6 (lib) (requires download & build)
 - cabal-doctest-1.0.9 (lib) (requires download & build)
 - indexed-traversable-0.1.2 (lib) (requires download & build)
 - th-abstraction-0.4.3.0 (lib) (requires download & build)
 - distributive-0.6.2.1 (lib) (requires download & build)
 - comonad-5.0.7 (lib:comonad) (requires download & build)
 - bifunctors-5.5.11 (lib) (requires download & build)
 - assoc-1.0.2 (lib) (requires download & build)
 - these-1.1.1.1 (lib) (requires download & build)
 - strict-0.4.0.1 (lib) (requires download & build)
Downloading  base-orphans-0.8.6
Downloaded   base-orphans-0.8.6
Downloading  th-abstraction-0.4.3.0
Starting     base-orphans-0.8.6 (lib)
Downloaded   th-abstraction-0.4.3.0
Downloading  distributive-0.6.2.1
Starting     th-abstraction-0.4.3.0 (lib)
Downloaded   distributive-0.6.2.1
Downloading  indexed-traversable-0.1.2
Downloaded   indexed-traversable-0.1.2
Downloading  cabal-doctest-1.0.9
Starting     indexed-traversable-0.1.2 (lib)
Downloaded   cabal-doctest-1.0.9
Downloading  comonad-5.0.7
Starting     cabal-doctest-1.0.9 (lib)
Downloaded   comonad-5.0.7
Downloading  bifunctors-5.5.11
Downloaded   bifunctors-5.5.11
Downloading  assoc-1.0.2
Downloaded   assoc-1.0.2
Downloading  these-1.1.1.1
Downloaded   these-1.1.1.1
Downloading  strict-0.4.0.1
Downloaded   strict-0.4.0.1
Building     cabal-doctest-1.0.9 (lib)
Building     indexed-traversable-0.1.2 (lib)
Building     base-orphans-0.8.6 (lib)
Building     th-abstraction-0.4.3.0 (lib)

Failed to build base-orphans-0.8.6.
Build log (
/home/stalbert/snap/clash/1215/.cabal/logs/ghc-8.6.5/base-orphans-0.8.6-6810165c18561db86da07dec9947c217410c7d33ff034e9aa94440cf7aa1ea8e.log
):
Configuring library for base-orphans-0.8.6..
Preprocessing library for base-orphans-0.8.6..
Building library for base-orphans-0.8.6..
[1 of 2] Compiling Data.Orphans.Prelude ( src/Data/Orphans/Prelude.hs, dist/build/Data/Orphans/Prelude.o )
[2 of 2] Compiling Data.Orphans     ( src/Data/Orphans.hs, dist/build/Data/Orphans.o )
/snap/clash/1215/usr/bin/ld.gold: error: cannot open /snap/clash/1215/usr/bin/../lib/gcc/x86_64-linux-gnu/9/../../../x86_64-linux-gnu/libgmp.so: No such file or directory
collect2: error: ld returned 1 exit status
`x86_64-linux-gnu-gcc' failed in phase `Linker'. (Exit code: 1)

Failed to build cabal-doctest-1.0.9.
Build log (
/home/stalbert/snap/clash/1215/.cabal/logs/ghc-8.6.5/cabal-doctest-1.0.9-9fcf36e3351ed87109c1a834ad0b5d492d0dab4a93c52a23b7654b5a6f22bce1.log
):
Configuring library for cabal-doctest-1.0.9..
Preprocessing library for cabal-doctest-1.0.9..
Building library for cabal-doctest-1.0.9..
[1 of 1] Compiling Distribution.Extra.Doctest ( src/Distribution/Extra/Doctest.hs, dist/build/Distribution/Extra/Doctest.o )
/snap/clash/1215/usr/bin/ld.gold: error: cannot open /snap/clash/1215/usr/bin/../lib/gcc/x86_64-linux-gnu/9/../../../x86_64-linux-gnu/libgmp.so: No such file or directory
collect2: error: ld returned 1 exit status
`x86_64-linux-gnu-gcc' failed in phase `Linker'. (Exit code: 1)

Failed to build indexed-traversable-0.1.2.
Build log (
/home/stalbert/snap/clash/1215/.cabal/logs/ghc-8.6.5/indexed-traversable-0.1.2-eace64dd74f538ba2bdc445d289d87e28ec3e56f64fb75515102e6a6ce6833b6.log
):
Configuring library for indexed-traversable-0.1.2..
Preprocessing library for indexed-traversable-0.1.2..
Building library for indexed-traversable-0.1.2..
[1 of 5] Compiling GhcExts          ( src/GhcExts.hs, dist/build/GhcExts.o )
[2 of 5] Compiling WithIndex        ( src/WithIndex.hs, dist/build/WithIndex.o )
[3 of 5] Compiling Data.Traversable.WithIndex ( src/Data/Traversable/WithIndex.hs, dist/build/Data/Traversable/WithIndex.o )
[4 of 5] Compiling Data.Functor.WithIndex ( src/Data/Functor/WithIndex.hs, dist/build/Data/Functor/WithIndex.o )
[5 of 5] Compiling Data.Foldable.WithIndex ( src/Data/Foldable/WithIndex.hs, dist/build/Data/Foldable/WithIndex.o )
/snap/clash/1215/usr/bin/ld.gold: error: cannot open /snap/clash/1215/usr/bin/../lib/gcc/x86_64-linux-gnu/9/../../../x86_64-linux-gnu/libgmp.so: No such file or directory
collect2: error: ld returned 1 exit status
`x86_64-linux-gnu-gcc' failed in phase `Linker'. (Exit code: 1)

Failed to build th-abstraction-0.4.3.0.
Build log (
/home/stalbert/snap/clash/1215/.cabal/logs/ghc-8.6.5/th-abstraction-0.4.3.0-eec533fe824406df782d108cbff2047ea026e5017fbdeba684dfd1da1d0dea47.log
):
Configuring library for th-abstraction-0.4.3.0..
Preprocessing library for th-abstraction-0.4.3.0..
Building library for th-abstraction-0.4.3.0..
[1 of 3] Compiling Language.Haskell.TH.Datatype.Internal ( src/Language/Haskell/TH/Datatype/Internal.hs, dist/build/Language/Haskell/TH/Datatype/Internal.o )
[2 of 3] Compiling Language.Haskell.TH.Datatype.TyVarBndr ( src/Language/Haskell/TH/Datatype/TyVarBndr.hs, dist/build/Language/Haskell/TH/Datatype/TyVarBndr.o )
[3 of 3] Compiling Language.Haskell.TH.Datatype ( src/Language/Haskell/TH/Datatype.hs, dist/build/Language/Haskell/TH/Datatype.o )
/snap/clash/1215/usr/bin/ld.gold: error: cannot open /snap/clash/1215/usr/bin/../lib/gcc/x86_64-linux-gnu/9/../../../x86_64-linux-gnu/libgmp.so: No such file or directory
collect2: error: ld returned 1 exit status
`x86_64-linux-gnu-gcc' failed in phase `Linker'. (Exit code: 1)
cabal-3.2: Failed to build base-orphans-0.8.6 (which is required by
strict-0.4.0.1). See the build log above for details.
Failed to build cabal-doctest-1.0.9 (which is required by strict-0.4.0.1). See
the build log above for details.
Failed to build indexed-traversable-0.1.2 (which is required by
strict-0.4.0.1). See the build log above for details.
Failed to build th-abstraction-0.4.3.0 (which is required by strict-0.4.0.1).
See the build log above for details.
```

Sure enough, there are several broken symlinks in /snap/clash/1215/usr/lib/x86_64-linux-gnu:
```
$ find /snap/clash/1215/usr/lib/x86_64-linux-gnu -xtype l
/snap/clash/1215/usr/lib/x86_64-linux-gnu/libbsd.so
/snap/clash/1215/usr/lib/x86_64-linux-gnu/libffi.so
/snap/clash/1215/usr/lib/x86_64-linux-gnu/libform.so
/snap/clash/1215/usr/lib/x86_64-linux-gnu/libformw.so
/snap/clash/1215/usr/lib/x86_64-linux-gnu/libgmp.so
/snap/clash/1215/usr/lib/x86_64-linux-gnu/libmenu.so
/snap/clash/1215/usr/lib/x86_64-linux-gnu/libmenuw.so
/snap/clash/1215/usr/lib/x86_64-linux-gnu/libpanel.so
/snap/clash/1215/usr/lib/x86_64-linux-gnu/libpanelw.so
/snap/clash/1215/usr/lib/x86_64-linux-gnu/libtic.so
/snap/clash/1215/usr/lib/x86_64-linux-gnu/libyaml.so
```

--- a/.ci/bindist/linux/snap/snap/snapcraft.yaml
+++ b/.ci/bindist/linux/snap/snap/snapcraft.yaml
@@ -39,6 +39,15 @@ parts:
       # To save time on CI, we currently don't build profiling packages anymore. We should
       # probably selectively build them on release branches and nightlies.
       #- clash-prof
+      # The below packages are included in core20, but their -dev packages are
+      # not.  Include them here to avoid broken symlinks from dev sonames.
+      - libbsd0
+      - libffi7
+      - libgmp10
+      - libncurses6
+      - libncursesw6
+      - libtinfo6
+      - libyaml-0-2
     override-prime: |
       snapcraftctl prime
       apt-get install -y ghc

